pest/src/span.rs

```diff
@@ -176,7 +201,7 @@ impl<'i> Eq for Span<'i> {}
 
 impl<'i> Hash for Span<'i> {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        (self.input as *const [u8]).hash(state);
+        (self.input.as_bytes() as *const [u8]).hash(state);
```

Comment:
I believe this approach only hashes the first hash of the `&str`/fat pointer.

```suggestion
        (self.input as *const str).hash(state);
```